src/cream.erl

```diff
@@ -54,8 +54,8 @@ new(MaxCapacity, CacheOpts) ->
 cache(Cache, Key, ExpensiveValFun) ->
     KeyBin = term_to_binary(Key),
     case cream_nif:get(Cache, KeyBin) of
-        {ok, CachedVal} ->
-            CachedVal;
+        {ok, CachedValBin} ->
+            binary_to_term(CachedValBin);
         notfound ->
             Val = ExpensiveValFun(),
             ok = cream_nif:insert(Cache, KeyBin, Val),
```

Comment:
`cream_nif:insert` expects `Val` is a binary. Probably want to wrap `Val` in `term_to_binary/1` here.